agent/xds/listeners.go

```diff
@@ -270,6 +354,46 @@ func (s *Server) makePublicListener(cfgSnap *proxycfg.ConfigSnapshot, token stri
 	return l, err
 }
 
+func (s *Server) makeExposedCheckListener(cfgSnap *proxycfg.ConfigSnapshot, cluster, path, protocol string, port int) (proto.Message, error) {
+	cfg, err := ParseProxyConfig(cfgSnap.Proxy.Config)
+	if err != nil {
+		// Don't hard fail on a config typo, just warn. The parse func returns
+		// default config if there is an error so it's safe to continue.
+		s.Logger.Printf("[WARN] envoy: failed to parse Connect.Proxy.Config: %v", err)
+	}
+
+	// No user config, use default listener
+	addr := cfgSnap.Address
+
+	// Override with bind address if one is set, otherwise default to 0.0.0.0
+	if cfg.BindAddress != "" {
+		addr = cfg.BindAddress
+	} else if addr == "" {
+		addr = "0.0.0.0"
+	}
+
+	// Strip any special characters from path
```

Comment:
```suggestion
	// Strip any special characters from path to make a valid and hopefully unique name
```
I think this is fine but technically we could end up with multiple listeners here with same name if paths only differ in their special chars.